br/pkg/lightning/importer/tidb.go

```diff
@@ -279,7 +278,34 @@ func ObtainImportantVariables(ctx context.Context, g glue.SQLExecutor, needTiDBV
 		}
 	}
 	query.WriteString("')")
-	kvs, err := g.QueryStringsWithLog(ctx, query.String(), "obtain system variables", log.FromContext(ctx))
+	kvs := make([][]string, 0, len(common.DefaultImportantVariables)+len(common.DefaultImportVariablesTiDB))
+	exec := common.SQLWithRetry{DB: db, Logger: log.FromContext(ctx)}
+	err := exec.Transact(ctx, "obtain system variables", func(c context.Context, tx *sql.Tx) (txErr error) {
```

Comment:
make it a function?

Reply:
do you mean provide a function like `common.TransactWithRetry(ctx, db, logger, purpose, actionFn)`? We still need provide these arguments, I think there's not much benefits

Reply:
like previous `QueryStringsWithLog`, we can move it to `SQLWithRetry`

Reply:
https://github.com/pingcap/tidb/pull/42911/commits/a3288912a339009706e538378037847a7fd71b8a